pkg/pod/pod_test.go

```diff
@@ -780,6 +884,12 @@ script-heredoc-randomly-generated-78c5n
 			if d := cmp.Diff(c.want, &got.Spec, resourceQuantityCmp); d != "" {
 				t.Errorf("Diff(-want, +got):\n%s", d)
 			}
+
+			if c.wantAnnotations != nil {
```

Comment:
One thing I realized while adding the tests, is that we were never using the wantAnnotations filed at all (it was only specified for the sidecar tests). I added this block and started ignoring the releaseAnnotation that gets added by default.

Reply:
thanks for catching this! might also be a sign that the test cases are a bit too complex